### PR TITLE
Add acp example

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    TransactionBuilder txBuilder = new TransactionBuilder(api);
+    IndexerCollector txUtils = new IndexerCollector(api, ckbIndexerApi);
+
+    List<Receiver> receivers =
+        Collections.singletonList(new Receiver(receiverAcpAddress, Utils.ckbToShannon(200)));
+    List<CellOutput> cellOutputs = txUtils.generateOutputs(receivers, SendAddresses.get(0));
+    txBuilder.addOutputs(cellOutputs);
+
+    txBuilder.setOutputsData(Arrays.asList("0x", "0x"));
```
Comment (shaojunda) [inline example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:86]: There is no need for manual assignment, right? https://github.com/nervosnetwork/ckb-sdk-java/blob/11531e0f4b94de898553f8e01e0c7f278d914c65/ckb/src/main/java/org/nervos/ckb/transaction/TransactionBuilder.java#L94

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        "Before transferring, first sender's balance: "
+            + getBalance(SendAddresses.get(0)).divide(UnitCKB).toString(10)
+            + " CKB");
+
+    String acpHash = createACPCell();
+    System.out.println("Create acp cell tx hash: " + acpHash);
+
+    // waiting transaction into block, sometimes you should wait more seconds
+    Thread.sleep(30000);
+    System.out.println("Transfer SUDT tx hash: " + transfer(new OutPoint(acpHash, "0x0")));
```
Comment (shaojunda) [inline example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:68]: there should be Transfer `CKB`?

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    for (ScriptGroupWithPrivateKeys scriptGroupWithPrivateKeys : scriptGroupWithPrivateKeysList) {
+      signBuilder.sign(
+          scriptGroupWithPrivateKeys.scriptGroup, scriptGroupWithPrivateKeys.privateKeys.get(0));
+    }
+    Transaction tx = signBuilder.buildTx();
+    return api.sendTransaction(tx);
+  }
+
+  private static String transfer(OutPoint acpOutPoint) throws IOException {
```
Comment (shaojunda) [inline example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:130]: If the `transfer` function accepts an `outpoint`, We may need to add an example of collecting the ACP outpoint.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    CollectResult collectResult =
+        txUtils.collectInputs(SendAddresses, txBuilder.buildTx(), feeRate, Sign.SIGN_LENGTH * 2);
+
+    // update change cell output capacity after collecting cells if there is changeOutput
+    if (Numeric.toBigInt(collectResult.changeCapacity).compareTo(MIN_CKB) >= 0) {
+      cellOutputs.get(cellOutputs.size() - 1).capacity = collectResult.changeCapacity;
+      txBuilder.setOutputs(cellOutputs);
+    }
+
+    txBuilder.setOutputsData(Arrays.asList("0x", "0x"));
```
Comment (shaojunda) [inline example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:157]: If there is no change output, there should be only one output data.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+import org.nervos.ckb.utils.Numeric;
+import org.nervos.ckb.utils.Utils;
+import org.nervos.ckb.utils.address.AddressGenerator;
+import org.nervos.ckb.utils.address.AddressParser;
+
+/** Copyright © 2021 Nervos Foundation. All rights reserved. */
+// ACP RFC:
+// https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0026-anyone-can-pay/0026-anyone-can-pay.md
+public class ACPTransactionExample {
+  // ACP_CKB_MINIMUM is set to 9, which means in each transaction, one must at least transfers 10^9
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:32]: To reduce complexity, udt is not introduced in the example. But I can set `SUDT_MINIMUM` zero.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 3 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:130]: The `acpOutPoint` comes from the result of `createAcpCell` function.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 1 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:86]: No. For a simple transaction, the developer maybe does not care about the outputs_data, and he could not set outputs_data.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  // shannons, or 10 CKBytes into the anyone-can-pay cel
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java]: Updated https://github.com/nervosnetwork/ckb-sdk-java/pull/382/commits/fb6f3c18af1386b5a960b7f06af73f2a014d3d55

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:32]: Updated https://github.com/nervosnetwork/ckb-sdk-java/pull/382/commits/fb6f3c18af1386b5a960b7f06af73f2a014d3d55

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 2 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java]: Updated https://github.com/nervosnetwork/ckb-sdk-java/pull/382/commits/fb6f3c18af1386b5a960b7f06af73f2a014d3d55

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 4 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java]: Updated https://github.com/nervosnetwork/ckb-sdk-java/pull/382/commits/fb6f3c18af1386b5a960b7f06af73f2a014d3d55

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 3 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:132]: This is to reduce complexity

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 1 above)
Comment (shaojunda) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java]: Why do you manually set `outputs_data ` here?

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 3 above)
Comment (shaojunda) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java:132]: got it.

File: example/src/main/java/org/nervos/ckb/ACPTransactionExample.java (hunk 1 above)
Comment (duanyytop) [reply example/src/main/java/org/nervos/ckb/ACPTransactionExample.java]: If the developer forgets to set the `outputsData`, I will help him to fix it

**Review by shaojunda — APPROVED** [review]

**Pull request merged**